cogs/tags/tags.py

```diff
@@ -0,0 +1,1024 @@
+            e.add_field(name=emoji + ' Generic Tag', value=fmt.format(tag))
+
+        for emoji, tag in self.top_three_tags(db):
+            e.add_field(name=emoji + ' Server Tag', value=fmt.format(tag))
+
+        await ctx.send(embed=e)
+
+    @tag.command()
+    @role_or_mod_or_permissions(role=ALLOWED_ROLE, manage_messages=True)
+    async def edit(self, ctx: Context, name : str, *, content : str):
+        """Modifies an existing tag that you own.
+        This command completely replaces the original text. If you edit
+        a tag via private message then the tag is looked up in the generic
+        tag database. Otherwise it looks at the server-specific database.
+        """
+
+        content = self.clean_tag_content(content)
+        lookup = name.lower()
+        server = ctx.message.guild
+        try:
+            tag = self.get_tag(server, lookup, redirect=False)
+        except RuntimeError as e:
+            await ctx.send(e)
+            return
+
+        if isinstance(tag, TagAlias):
+            await ctx.send('Cannot edit tag aliases. Remake it if you want to re-point it.')
+            return
+
+        mod_roles = await self.bot.get_mod_roles(server)
+        admin_roles = await self.bot.get_admin_roles(server)
+        roles = ctx.author.roles
+        
+        # Check and see if the user is not the tag owner, or is not a mod, or is not an admin.
+        if (tag.owner_id != str(ctx.message.author.id) and
+                not list(set(admin_roles) & set(roles)) and
+                not list(set(mod_roles) & set(roles)) and
+                not await self.bot.is_owner(ctx.author)):
+            await ctx.send('Only the tag owner can edit this tag.')
+            return
+
+        db = self.config.get(tag.location)
+        tag.content = content
+        await self.config.put(tag.location, db)
+        await ctx.send('Tag successfully edited.')
+
+    @tag.command(name="transfer")
+    @commands.guild_only()
+    @role_or_mod_or_permissions(role=ALLOWED_ROLE, manage_messages=True)
+    async def transfer(self, ctx: Context, tag_name, user: discord.Member):
+        """Transfer your tag to another user.
+
+        This can be done by the creator of the tag. Cannot transfer 
+        if the user being transfered to is over the tag limit.
+
+        Parameters:
+        -----------
+        tag_name: str
+            The tag name.
+        user: discord.Member
+            The guild member you wish to transfer this tag to.
+        """
+        lookup = tag_name.lower().strip()
+        server = ctx.message.guild
+        try:
+            tag = self.get_tag(server, lookup, redirect=False)
+        except RuntimeError as e:
+            await ctx.send(e)
+            return
+
+        mod_roles = await self.bot.get_mod_roles(server)
+        admin_roles = await self.bot.get_admin_roles(server)
+
+        sensei = discord.utils.get(ctx.message.guild.roles, name=ALLOWED_ROLE)
```

Comment:
Nitpick but this should probably be settable.

Reply:
I'll open an issue for this to track, will fix in future PR.

Reply:
Tracked by #292